esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -509,6 +515,10 @@ void EsdlServiceImpl::handleServiceRequest(IEspContext &context,
         }
         else
         {
+            context.addTraceSummaryTimeStamp(LogNormal, "srt-reqtrans");
+            m_customRequestTransform->processTransform(&context, req, m_oEspBindingCfg.get());
```

Comment:
The req at this point hasn't been transformed in any way, applying the transform here allows clean xpath mapping into the req... but new fields resulting from the transform are subject for filtering in the esdl transform process line 527. Considering moving this transform post esdl transform. the request at that point will be wrapped in a row element...

Reply:
@rpastrana as we discussed, and as mentioned above this should happen after the built in ESDL transform.  This custom transform can be considered part of the service and should operate on already versioned data.  Also this transform should be able to add extra information for the next hop (usually Roxie) that isn't necessarily in the ESDL.  In the future we can consider adding some form of validation, etc.